packages/fast-usdc/src/exos/liquidity-pool.js

```diff
@@ -0,0 +1,232 @@
+import {
+  AmountMath,
+  AmountShape,
+  PaymentShape,
+  RatioShape,
+} from '@agoric/ertp';
+import {
+  makeRecorderTopic,
+  TopicsRecordShape,
+} from '@agoric/zoe/src/contractSupport/topics.js';
+import { depositToSeat } from '@agoric/zoe/src/contractSupport/zoeHelpers.js';
+import { SeatShape } from '@agoric/zoe/src/typeGuards.js';
+import { M } from '@endo/patterns';
+import { Fail, q } from '@endo/errors';
+import {
+  depositCalc,
+  makeParity,
+  withdrawCalc,
+  withFees,
+} from '../pool-share-math.js';
+import { makeProposalShapes } from '../type-guards.js';
+
+/**
+ * @import {Zone} from '@agoric/zone';
+ * @import {Remote, TypedPattern} from '@agoric/internal'
+ * @import {StorageNode} from '@agoric/internal/src/lib-chainStorage.js'
+ * @import {MakeRecorderKit, RecorderKit} from '@agoric/zoe/src/contractSupport/recorder.js'
+ * @import {USDCProposalShapes, ShareWorth} from '../pool-share-math.js'
+ */
+
+const { add, isEqual } = AmountMath;
+
+/** @param {Brand} brand */
+const makeDust = brand => AmountMath.make(brand, 1n);
+
+/**
+ *
+ * @param {ZCFSeat} poolSeat
+ * @param {ShareWorth} shareWorth
+ * @param {Brand} USDC
+ */
+const checkPoolBalance = (poolSeat, shareWorth, USDC) => {
+  const available = poolSeat.getAmountAllocated('USDC', USDC);
+  const dust = makeDust(USDC);
+  isEqual(add(available, dust), shareWorth.numerator) ||
+    Fail`🚨 pool balance ${q(available)} inconsistent with shareWorth ${q(shareWorth)}`;
+};
+
+/**
+ * @param {Zone} zone
+ * @param {ZCF} zcf
+ * @param {Brand<'nat'>} USDC
+ * @param {{
+ *   makeRecorderKit: MakeRecorderKit;
+ * }} tools
+ */
+export const prepareLiquidityPoolKit = (zone, zcf, USDC, tools) => {
+  return zone.exoClassKit(
+    'Fast Liquidity Pool',
```

Comment:
an exo kind is always per vat, right? no need to include `Fast`.

This is also pretty close to being a general capability for managing a liquidity pool, especially if you move out the public interface.